src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -470,11 +481,29 @@ public boolean isEnqueued() {
     }
 
     /**
-     * Clears this reference object and adds it to the queue with which
-     * it is registered, if any.
+     * Clears this reference object, then attempts to add it to the queue with
+     * which it is registered, if any.
      *
-     * <p> This method is invoked only by Java code; when the garbage collector
+     * <p>If this reference was already enqueued (by the garbage collector, or a
+     * previous call to {@code enqueue}), this method is <b><i>not successful</i></b>,
+     * and returns false.
```

Comment:
If we're going to talk about successful and unsuccessful calls, we should define what success is first. I guess that would be something like: if this ref is registered with a queue and not already enqueued, it is enqueued successfully and the method returns true. Otherwise, not successful, and returns false.

Reply:
This could be worded as a post condition, after the call the ref is enqueued with the queue; the return is true iff the ref was not previously enqueued.
The enqueuing is not conditional (assuming the queue is non-null).

Reply:
I'll give that some thought. The enqueuing is not conditional, but the _happens-before_ is...